rust/lance-core/src/utils/mask.rs

```diff
@@ -328,6 +334,81 @@ impl RowIdTreeMap {
         }
     }
 
+    /// Insert a single value into the set
+    ///
+    /// Returns true if the value was not already in the set.
+    ///
+    /// ```rust
+    /// use lance_core::utils::mask::RowIdTreeMap;
+    ///
+    /// let mut set = RowIdTreeMap::new();
+    /// assert_eq!(set.insert(10), true);
+    /// assert_eq!(set.insert(10), false);
+    /// assert_eq!(set.contains(10), true);
+    /// ```
+    pub fn insert(&mut self, value: u64) -> bool {
+        let fragment = (value >> 32) as u32;
+        let row_addr = value as u32;
+        match self.inner.get_mut(&fragment) {
+            None => {
+                let mut set = RoaringBitmap::new();
+                set.insert(row_addr);
+                self.inner.insert(fragment, RowIdSelection::Partial(set));
+                true
+            }
+            Some(RowIdSelection::Full) => false,
+            Some(RowIdSelection::Partial(set)) => set.insert(row_addr),
+        }
+    }
+
+    /// Insert a range of values into the set
+    pub fn insert_range<R: RangeBounds<u64>>(&mut self, range: R) -> u64 {
+        let (mut start_high, mut start_low) = match range.start_bound() {
```

Comment:
I'm a little confused here.  `start_high` / `start_low` are `u32`.  Are they fragment IDs?  Or row offsets?

Reply:
As we move to generic u64 row ids, the high and low bits no longer consistently have the semantics of fragment ids and row offsets. Thus I rename them "high" and "low".